vault/request_handling.go

```diff
@@ -364,6 +363,26 @@ func (c *Core) HandleRequest(httpCtx context.Context, req *logical.Request) (res
 		}
 	}(ctx, httpCtx)
 
+	ns, err := namespace.FromContext(httpCtx)
+	if err != nil {
+		cancel()
+		c.stateLock.RUnlock()
+		return nil, errwrap.Wrapf("could not parse namespace from http context: {{err}}", err)
+	}
+	ctx = namespace.ContextWithNamespace(ctx, ns)
+
+	resp, err = c.handleCancelableRequest(ctx, ns, req)
+
+	if req != nil {
```

Comment:
If req can be nil, we probably shouldn't be invoking c.handleCancelableRequest with it.  Will panic when trying to access req.Path.

Reply:
Cool